src/tess.py

```diff
@@ -0,0 +1,359 @@
+        )
+        if not any(close_mags):
+            self._print("No objects found within mag difference "
+                        "threshold for TIC query.", verbose)
+            return pd.DataFrame()
+
+        cat_data_dists = np.array(catalog_data['dstArcSec'])
+        close_dsts = np.logical_and(
+            ~np.isnan(cat_data_dists),
+            cat_data_dists < self.pars.distance_thresh
+        )
+        if not any(close_dsts):
+            self._print("No objects found within distance "
+                        "threshold for TIC query.", verbose)
+            return pd.DataFrame()
+
+        close_mags_and_dsts = np.logical_and(close_mags, close_dsts)
+        cand_count = 0
+        for b in close_mags_and_dsts:
+            if b: cand_count += 1
+        # if there are multiple likely candidates, throw warning
+        if cand_count > 1:
+            warnings.warn(f"Multiple ({close_mags}) sources with similar "
+                            "magnitudes in cone search. "
+                            "Might select incorrect source.")
+
+        candidate_idx = np.nanargmin(cat_data_dists)
+        tess_name = 'TIC ' + catalog_data['ID'][candidate_idx]
+        obs_params = {
+            'objectname': tess_name,
+            'radius': self.pars.obs_qry_radius,
+            'obs_collection': 'TESS',
+        }
+        data_query = self._try_query(Observations.query_criteria, obs_params, verbose)
+        if len(data_query) == 0:
+            self._print(f"No data found for object {tess_name}.", verbose)
+            return pd.DataFrame()
+        if 'timeseries' not in data_query['dataproduct_type']:
+            self._print(
+                f"No timeseries data found for object {tess_name}.", verbose)
+            return pd.DataFrame()
+        
+        lc_indices = []
+        for i, uri in enumerate(data_query['dataURL']):
+            if isinstance(uri, str) and uri[-7:-5] == 'lc':
+                lc_indices.append(i)
+
+        if not lc_indices:
+            self._print(f"No lightcurve data found for object {tess_name}.", verbose)
+            return pd.DataFrame()
+
+        self._print(f"Found {len(lc_indices)} light curve(s) "
+                     "for this source.", verbose)
+
+        base_url = "https://mast.stsci.edu/api/v0.1/Download/file?uri="
+
+        header_attributes_to_save = [
+            'TICVER', 'RA_OBJ', 'DEC_OBJ',
+            'PMRA', 'PMDEC', 'TESSMAG',
+            'TEFF', 'LOGG', 'MH',
+            'RADIUS', 'CRMITEN',
+            'CRSPOC', 'CRBLKSZ',
+        ]
+        df_list = []
+        altdata = {}
+        altdata['source mag'] = mag
+        altdata['source coordinates'] = coord_str
+        sectors = set()
+        TICID = set()
+        for i in lc_indices:
+            uri = data_query['dataURL'][i]
+            table = self._try_table_read_fits(base_url + uri)
+            hdul0header, hdul2data = self._try_open_fits(base_url + uri)
```

Comment:
Are these two separate calls to the API? That seems like a big waste of resources...

Reply:
I'll try to make it only one call